src/services/helperFunctionsServices.ts

```diff
@@ -0,0 +1,306 @@
+
+    // text filters (case-insensitive contains)
+    if (filters.nameEn) {
+        query = query.ilike('nameEn', `%${filters.nameEn}%`) as B
+    }
+    if (filters.nameJa) {
+        query = query.ilike('nameJa', `%${filters.nameJa}%`) as B
+    }
+
+    return query
+}
+
+/** 
+*===================================
+*- HP section helpers function
+*===================================
+*/
+
+// Builds a partial update patch for Healthcare Professional rows.
+export function buildHpUpdatePatch(fields: Partial<gqlTypes.UpdateHealthcareProfessionalInput>) {
+    const updatePatch: Partial<dbSchema.DbHealthcareProfessionalRow> = {}
+
+    if (fields.names !== null) { updatePatch.names = fields.names }
+    if (fields.degrees !== null) { updatePatch.degrees = fields.degrees }
+    if (fields.spokenLanguages !== null) { updatePatch.spokenLanguages = fields.spokenLanguages }
+    if (fields.specialties !== null) { updatePatch.specialties = fields.specialties }
+    if (fields.acceptedInsurance !== null) { updatePatch.acceptedInsurance = fields.acceptedInsurance }
+    if (fields.additionalInfoForPatients !== undefined) {
+        updatePatch.additionalInfoForPatients = fields.additionalInfoForPatients
+    }
+    updatePatch.updatedDate = new Date().toISOString()
+    return updatePatch
+}
+
+// Applies JSONB array filters to an HP query builder (degrees, specialties, languages, insurance).
+export function applyHpFilters<B extends HasContains>(
+  builder: B,
+  filters: gqlTypes.HealthcareProfessionalSearchFilters
+): B {
+    let query = builder
+
+    if (filters.degrees?.length) { query = query.contains('degrees', filters.degrees as gqlTypes.Degree[]) as B }
+    if (filters.specialties?.length) { query = query.contains('specialties', filters.specialties as gqlTypes.Specialty[]) as B }
+    if (filters.spokenLanguages?.length) { query = query.contains('spokenLanguages', filters.spokenLanguages as gqlTypes.Locale[]) as B }
+    if (filters.acceptedInsurance?.length) { query = query.contains('acceptedInsurance', filters.acceptedInsurance as gqlTypes.Insurance[]) as B }
+    return query
+}
+
+// Maps GQL Create input → DB insert row; defaults arrays to [] to avoid `!`.
+export function mapCreateInputToHpInsertRow(
+  input: gqlTypes.CreateHealthcareProfessionalInput
+): dbSchema.HealthcareProfessionalInsertRow {
+    return {
+        names: input.names,
+        degrees: input.degrees ?? [],
+        spokenLanguages: input.spokenLanguages ?? [],
+        specialties: input.specialties ?? [],
+        acceptedInsurance: input.acceptedInsurance ?? [],
+        additionalInfoForPatients: input.additionalInfoForPatients ?? null,
+        createdDate: new Date().toISOString(),
+        updatedDate: new Date().toISOString()
+    }
+}
+
+// Derives the facilityId to associate from relationship edits (create/delete).
+export function resolveFacilityIdFromRelationships(
+  relationss: gqlTypes.Relationship[] | null | undefined
```

Comment:
```suggestion
  relations: gqlTypes.Relationship[] | null | undefined
```
Is this a typo? If so, it should be updated everywhere.